custom_components/opnsense/update.py

```
@@ -263,71 +204,90 @@
         running = True
         while running:
-            time.sleep(sleep_time)
+            await asyncio.sleep(sleep_time)
             try:
-                response = client.upgrade_status()
+                response = self._client.upgrade_status()
                 # after finished status is "done"
-                running = response["status"] == "running"
-            except:
+                running: bool = response["status"] == "running"
+            except Exception as e:
```

Ruff

Do not catch blind exception: `Exception`

(BLE001)